docs/book/v2/intro.md

```diff
@@ -0,0 +1,142 @@
+# Introduction
+
+This library provides the ability to dispatch middleware pipelines in place of
+controllers within laminas-mvc.
+
+## Dispatching PSR-7 Middleware
+
+[PSR-7](http://www.php-fig.org/psr/psr-7/) defines interfaces for HTTP messages,
+and is now being adopted by many frameworks; Laminas itself offers a
+parallel microframework targeting PSR-7 with [Mezzio](https://docs.mezzio.dev/mezzio).
+What if you want to dispatch PSR-7 middleware from laminas-mvc?
+
+laminas-mvc currently uses [laminas-http](https://github.com/laminas/laminas-http)
+for its HTTP transport layer, and the objects it defines are not compatible with
+PSR-7, meaning the basic MVC layer does not and cannot make use of PSR-7
+currently.
+
+However, starting with version 2.7.0, laminas-mvc offers
+`Laminas\Mvc\MiddlewareListener`. This `Laminas\Mvc\MvcEvent::EVENT_DISPATCH`
+listener listens prior to the default `DispatchListener`, and executes if the
+route matches contain a "middleware" parameter, and the service that resolves to
+is callable. When those conditions are met, it uses the [PSR-7 bridge](https://github.com/laminas/laminas-psr7bridge)
+to convert the laminas-http request and response objects into PSR-7 instances, and
+then invokes the middleware.
+
+Starting with laminas-mvc version 3.2.0, `Laminas\Mvc\MiddlewareListener` is deprecated and replaced
+by `Laminas\Mvc\Middleware\MiddlewareListener` provided by this package.  
+After package installation, `Laminas\Mvc\Middleware` module must be registered in your
+laminas-mvc based application.
+If the [`laminas-component-installer`](https://docs.laminas.dev/laminas-component-installer/)
+is installed, it will handle the module registration automatically.
+
+## Mapping routes to Middleware
```

Comment:
For titles and headlines we us the "Chicago Manual of Style" capitalization rules:

> 1. Capitalize the first and the last word.
> 2. Capitalize nouns, pronouns, adjectives, verbs, adverbs, and subordinate conjunctions.
> 3. Lowercase articles (a, an, the), coordinating conjunctions, and prepositions.
> 4. Lowercase the ‘to’ in an infinitive (I want to play guitar).

You do not need to know the rules, because this website helps you: https://capitalizemytitle.com/style/Chicago/